cmd/utils.go

```diff
@@ -482,19 +482,16 @@ func getProjectConfig(config *RootCommandConfig) (*ProjectConfig, error) {
 		stack := v.GetString("stack")
 		config.Debug.log("Project stack from config file: ", projectConfig.Stack)
 		imageRepo := config.CliConfig.GetString("images")
-		config.Debug.log("Image repository set to: ", imageRepo)
```

Comment:
Can you justify all of these changes to `utils.go`? I don't see why they are necessary.

Reply:
The `if len(imageComponents) < 3` no longer makes sense as this would never be the case anymore. This part overrides the project stack registry with "images" from CliConfig. I'm not sure why we would want this to be the case and I think we can remove `OverrideStackRegistry` altogether?

Reply:
I have tested it without `OverrideStackRegistry` and `init` without overriding the project stack registry with "images" works now. I will push the changes. If you test it and it works fine for you I can remove the `OverrideStackRegistry` function altogether

Reply:
TLDR: Please undo all changes to utils.go from this PR.

Hi Sandy, I can see why you don't think this is necessary, but first you have to remember backward compatibility. There will be existing template tars out in the field that don't yet fully qualify their stack registry. In those cases, we need this code to add the "images" registry. 
And second, `OverrideStackRegistry` is needed because of the new `--stack-registry` flag on `appsody run`. It will win over all other registries.

Reply:
Okay I will undo these changes and push them (as this might be from my lack of understanding?). But without these changes, if you do an `appsody stack package` and then an `appsody init`, the `--image-registry` value you set when doing the `appsody stack package` would just get replaced with `docker.io`...

Reply:
Okay, this was the justification I originally asked for. I agree this is bad behavior. So we have to figure out a way to preserve the OverrideStackRegistry flag behavior, but not default to docker.io. Let me take a closer look and get back to you.

Reply:
Alright. I reproduced the behavior and I think the proper fix is to reset `config.StackRegistry` during `init`. 

I hope you don't mind, I've pushed a commit to your branch (this PR) to fix it.

Please pull and test this now. I'm going to approve it, so if your testing passes, feel free to merge.

Reply:
I don't mind at all! Thank you!